addons/point_of_sale/models/pos_order.py

```diff
@@ -194,7 +194,7 @@ def _prepare_invoice_line(self, order_line):
             'quantity': order_line.qty if self.amount_total >= 0 else -order_line.qty,
             'discount': order_line.discount,
             'price_unit': order_line.price_unit,
-            'name': order_line.product_id.display_name,
+            'name': order_line.full_product_name or order_line.product_id.display_name,
```

Comment:
When I read the code, I see that you put the value "name" of SO line  in the description field of pos line. And here, what you are taking is the full_product_name which is a concatenation of the product name and the description. Is that what it is done when we create an invoice from SO? I've checked and it seems that it is only the content of "name" of SO that is used for the name in invoice.

Maybe instead of storing the value in "description", we should immediately set it in "full_product_name"

Reply:
Maybe I didn't understand what you wanted to do exactly, but when putting the description in the "full_product_name" it's showing the description as the product name in the PoS.